codegen/src/main/java/io/sundr/codegen/functions/ClassTo.java

```diff
@@ -191,6 +197,7 @@ public TypeDef apply(Class item) {
 
       Set<Class> copy = new HashSet<>(references);
       copy.stream()
+          .peek(c -> references.remove(c))
```

Comment:
Why is remove called here, when it's also called in `forEach`?

Reply:
This was meant to replace the remove in forEach, so that we always remove items, even filtered ones.

Reply:
It doesn't sound like a good idea to have side effects in a call to `peek`…

Reply:
why?

Reply:
See the `peek` javadoc :)

Reply:
Sure, but that doesn't mean that I have to duplicate the loop just because something is marked `mostly for logging`.